Compiler Error from cmake-package

Hi,
I am trying to use cmake-package, but I am getting compiler erros from inside the cmake-package crate:

```shell
error[E0423]: expected function, found built-in attribute `link_name`
   --> ~/.cargo/git/checkouts/cmake-package-rs-6a32c4ccd67826e9/664e077/src/lib.rs:273:63
    |
273 |                 writeln!(io, "cargo:rustc-link-lib=dylib={}", link_name(lib)).unwrap();
    |                                                               ^^^^^^^^^ not a function

error[E0425]: cannot find function `link_dir` in this scope
   --> ~/.cargo/git/checkouts/cmake-package-rs-6a32c4ccd67826e9/664e077/src/lib.rs:276:32
    |
276 |             if let Some(lib) = link_dir(lib) {
    |                                ^^^^^^^^ not found in this scope

Some errors have detailed explanations: E0423, E0425.
For more information about an error, try `rustc --explain E0423`.
error: could not compile `cmake-package` (lib) due to 2 previous errors
```

Here is my `build.rs` if it helps:

```rust
use cmake_package::{find_cmake, find_package, Error, Version};

fn main() -> Result<(), Box<dyn std::error::Error>> {
    
    let quest_package = find_package("QuEST").find()?;
    let quest_target = quest_package.target("QuEST::QuEST")
        .ok_or("Failed to find QuEST::QuEST target")?;
    
    println!("QuEST: {:?}", quest_target);
    println!("QuEST compile definitions: {:?}", quest_target.compile_definitions);
    println!("QuEST compile options: {:?}", quest_target.compile_options);

    // Link the QuEST library
    quest_target.link();
    
    Ok(())
}
```

--- a/.github/workflows/ci.yaml
+++ b/.github/workflows/ci.yaml
@@ -17,7 +17,7 @@ jobs:
         os:
           - ubuntu-latest
           - windows-latest
-          #- macos-latest TODO
+          - macos-latest
       fail-fast: false
     env:
       CARGO_TERM_COLOR: always

--- a/.github/workflows/test-results.yaml
+++ b/.github/workflows/test-results.yaml
@@ -17,7 +17,7 @@ jobs:
         os:
           - ubuntu-latest
           - windows-latest
-          #- macos-latest TODO
+          - macos-latest
 
     runs-on: ubuntu-latest
     steps:

--- a/src/lib.rs
+++ b/src/lib.rs
@@ -81,9 +81,9 @@
 
 use std::io::Write;
 
-#[cfg(target_os = "linux")]
+#[cfg(any(target_os = "linux", target_os = "macos"))]
 use once_cell::sync::Lazy;
-#[cfg(target_os = "linux")]
+#[cfg(any(target_os = "linux", target_os = "macos"))]
 use regex::Regex;
 use tempfile::TempDir;
 
@@ -219,7 +219,7 @@ pub struct CMakeTarget {
 
 /// Turns /usr/lib/libfoo.so.5 into foo, so that -lfoo rather than -l/usr/lib/libfoo.so.5
 /// is passed to the linker. Leaves "foo" untouched.
-#[cfg(target_os = "linux")]
+#[cfg(any(target_os = "linux", target_os = "macos"))]
 fn link_name(lib: &str) -> &str {
     static RE: Lazy<Regex> = Lazy::new(|| Regex::new(r"lib([^/]+)\.so.*").unwrap());
     match RE.captures(lib) {
@@ -233,7 +233,7 @@ fn link_name(lib: &str) -> &str {
     lib
 }
 
-#[cfg(target_os = "linux")]
+#[cfg(any(target_os = "linux", target_os = "macos"))]
 fn link_dir(lib: &str) -> Option<&str> {
     static RE: Lazy<Regex> = Lazy::new(|| Regex::new(r"(.*)/lib[^/]+\.so.*").unwrap());
     RE.captures(lib)?.get(1).map(|f| f.as_str())
@@ -390,7 +390,7 @@ mod testing {
     }
 
     #[test]
-    #[cfg(target_os = "linux")]
+    #[cfg(any(target_os = "linux", target_os = "macos"))]
     fn test_link_to() {
         let target = CMakeTarget {
             name: "foo".into(),

--- a/src/lib.rs
+++ b/src/lib.rs
@@ -59,7 +59,7 @@
 //!
 //! # Known Limitations
 //!
-//! The crate currently supporst primarily linking against shared libraries. Linking against
+//! The crate currently supports primarily linking against shared libraries. Linking against
 //! static libraries is not tested and may not work as expected. The crate currently does not
 //! support linking against MacOS frameworks.
 //!

